core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -110,15 +110,25 @@ KOKKOS_IMPL_HOST_FUNCTION inline uint64_t clock_tic_host() noexcept {
 
   return ((uint64_t)a) | (((uint64_t)d) << 32);
 
-#elif defined(__powerpc) || defined(__powerpc__) || defined(__powerpc64__) || \
-    defined(__POWERPC__) || defined(__ppc__) || defined(__ppc64__)
+#elif defined(__powerpc64__) || defined(__ppc64__)
 
-  unsigned int cycles = 0;
+  unsigned long cycles = 0;
 
   asm volatile("mftb %0" : "=r"(cycles));
 
   return (uint64_t)cycles;
 
+#elif defined(__powerpc) || defined(__powerpc__) || defined(__POWERPC__) || \
+    defined(__ppc__)
+
+  unsigned int lower_word = 0;
+  unsigned int upper_word = 0;
+
+  asm volatile("mftb %0" : "=r"(lower_word));
+  asm volatile("mftbu %0" : "=r"(upper_word));
+
+  return (((uint64_t)upper_word) << 32 | ((uint64_t)lower_word));
+
 #else
 
   return std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Comment:
Separate from this PR, `high_resolution_clock` is not exactly deprecated, but its use is discouraged by knowledgeable folks: https://en.cppreference.com/w/cpp/chrono/high_resolution_clock

High quality software chronometry is unduly hard

Reply:
thanks for creating the issue